platform-api/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java

```diff
@@ -510,81 +511,106 @@ public String getFactorySnippet(@ApiParam(value = "Factory ID", required = true)
-                                                                        .path(vcs)
-                                                                        .path(workspace)
-                                                                        .path("import-source-descriptor")
-                                                                        .build().toString());
-            projectConfig = dtoFactory.createDto(ProjectConfigDto.class)
-                                      .withName(project.getName())
-                                      .withType(projectDescription.getType())
-                                      .withAttributes(attributes)
-                                      .withDescription(projectDescription.getDescription())
-                                      .withMixins(projectDescription.getMixins());
-
-            for (ModuleConfig module : projectManager.getProjectModules(project)) {
-                String moduleRelativePath = module.getPath().substring(project.getPath().length());
-
-                projectConfig.getModules().add(DtoFactory.newDto(ModuleConfigDto.class).withType(module.getType())
-                                                       .withPath(moduleRelativePath)
-                                                               //.withRecipe(moduleConfig.getRecipe())
-                                                       .withAttributes(module.getAttributes())
-                                                       .withMixins(module.getMixins())
-                                                       .withDescription(module.getDescription()));
+
+        Factory factory = DtoFactory.newDto(Factory.class)
+                                    .withWorkspace(asDto(usersWorkspace, path))
+                                    .withV("4.0");
+
+        return Response.ok(factory, MediaType.APPLICATION_JSON)
+                       .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=factory.json")
+                       .build();
+    }
+
+    private static WorkspaceConfigDto asDto(UsersWorkspaceImpl workspace, String projectPath) throws ConflictException {
+        final List<CommandDto> commands = workspace.getCommands()
+                                                   .stream()
+                                                   .map(DtoConverter::asDto)
+                                                   .collect(toList());
+
+        // Filter out projects by path and source storage presence.
+        Predicate<ProjectConfigImpl> predicate = new Predicate<ProjectConfigImpl>() {
+            @Override
+            public boolean test(ProjectConfigImpl projectConfig) {
+
+                if (projectPath != null) {
+                    if (!projectConfig.getPath().equals(projectPath)) {
+                        return false;
+                    }
+                }
+
+                return projectConfig.getSource() != null &&
```

Comment:
`&&` should be moved to a new line, see last statement from the [Lline wripping doc](https://github.com/codenvy/codenvy-dev-resources/wiki/Formatting#line-wrapping)